2_process.smk

```diff
@@ -0,0 +1,155 @@
+import os
+import numpy as np
+import pandas as pd
+
+configfile: "2_process/process_config.yaml"
+
+
+def mntoha_lake_sequence_files(wildcards):
+    """
+    List all MNTOHA lake sequence .npy files for training and testing.
+
+    :param wildcards: Snakemake wildcards (empty).
+    :returns: List of lake training/testing sequence files.
+
+    """
+    # make this function dependent on fetch_mntoha_metadata
+    # needed because lake_metadata.csv is used to determine lake_sequence_files
+    lake_metadata_file = checkpoints.fetch_mntoha_metadata.get(**wildcards).output[0]
+    lake_metadata = pd.read_csv(lake_metadata_file)
+    lake_sequence_files = [os.path.join('2_process/out/mntoha_sequences', f'sequences_{site_id}.npy') for site_id in lake_metadata.site_id]
+    return lake_sequence_files
+
+
+def save_sequences_summary(lake_sequence_files_input, summary_file):
+    """
+    Summarize the number of sequences with at least one temperature observation
+    for each lake, and save the result to csv
+
+    :param lake_sequence_files_input: the lake sequence files to summarize
+    :param summary_file: csv file with how many sequences are in each lake
+
+    """
+    sequence_counts = []
+    for sequences_file in lake_sequence_files_input:
+        num_sequences = np.load(sequences_file).shape[0] 
```

Comment:
Might just note here that the arrays saved in the sequence files have shape (# sequences, `sequence_length`, # depths + # features)